src/DynamoCoreWpf/ViewModels/Core/SerializationExtensions.cs

```diff
@@ -29,12 +29,13 @@ internal static string ToJson(this WorkspaceViewModel viewModel)
                 TypeNameHandling = TypeNameHandling.Auto,
                 Formatting = Formatting.Indented,
                 Converters = new List<JsonConverter>{
-                    new AnnotationViewModelConverter(),
-                    new NoteViewModelConverter(),
-                },
+                    new AnnotationViewModelConverter()
+                }
             };
 
+            viewModel.ConvertNotesToAnnotations();
```

Comment:
why do you need this? can't this be handled inside AnnotationViewModelConverter() ?

Reply:
I was expecting to use `AnnotationViewModelConverter` and overwrite `CanConvert` to include both types so we can write differently depending on type, do you think doing this way would help in format unification?

Reply:
This is necessary to be able to convert both notes and annotations as a single annotations collection. I initially attempted to handle it all in the AnnotationViewModelConverter but that ended up with an Annotations collection (as expected) but a Notes collection as well in the JSON. Doing it this way seemed cleaner to me and gave the expected result.

Reply:
why would you get a notes collection? you can control that in the converter, I guess.

Reply:
After adding a [JsonIgnore] attribute to the Notes collection there is no longer a Notes section being serialized in the JSON. I was forgetting about that code change, please ignore the comment about the Notes collection.

Reply:
The Annotations collection is serialized as a collection property of the WorkspaceViewModel object (on line 37 below). The AnnotationViewModelConverter serializes one AnnotationViewModel at a time, not the entire collection. In order to use the native JSON serialization methods it is necessary to have the notes as part of the annotation collection. In the converter you can control how an annotation is serialized, but not serializing multiple collections as a single collection (i.e. serializing notes and annotations as a single collection). I believe that this would require another higher-level converter and it seemed clearer to me to add the notes to the annotation collection, serialize, and then set the collections back to their previous state. I tried a few varieties of this and the methods that I added were the most straight-forward to me.

Reply:
My guess is you can use a different converter for this. Is this statement correct - Remove AnnotationVMConverter, instead add WorkspaceViewModelConverter, then start the write object, add annotations first, then notes. I am not sure how hard it will be.  The thing I am trying to get rid is adding - serializing - removing.

Reply:
I totally agree that the adding/serializing/removing is not the best solution, but I went that route because once there is a WorkspaceViewModelConverter it will be required to manually take care of all serialization for the workspace view model, not just the annotation special casing. I will take a look into what would be needed to go the new converter route, it is not clear to me right now how much that would involve.